operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventSource.java

```diff
@@ -176,7 +179,19 @@ public void onDelete(T resource, boolean b) {
     }
   }
 
+  public Stream<T> getCachedCustomResources() {
+    return getCachedCustomResources(a -> true);
+  }
 
+  public Stream<T> getCachedCustomResources(Predicate<T> predicate) {
```

Comment:
That seems like a very inefficient method.

Reply:
Changed it to a way how it's doable nicely with current java APIs but not sure it's actually much more efficient;